build.py

```diff
@@ -42,6 +43,22 @@ class ClusterVersionMappingError(RuntimeError):
     """An exception used to report cluster version mapping validation errors."""
 
 
+class RegexValidationError(RuntimeError):
+    """An exception used to report invalid regular expressions."""
+
+
+class GolangRegexValidator:
```

Comment:
Since we are using this class here, maybe we should move to another file and reuse it here and in the tests (`conftest.py` also has it).

Reply:
I was considering different options regarding this and decided to define the class separately here and in the tests,   at least for now. Since you approved this PR, I will go ahead and merge it. We can discuss it later.